src/lib.rs

```diff
@@ -602,6 +606,62 @@ impl<'t> Captures<'t> {
         self.named_groups.get(name).and_then(|i| self.get(*i))
     }
 
+    /// Expands all instances of `$group` in `replacement` to the corresponding
+    /// capture group `name`, and writes them to the `dst` buffer given.
+    ///
+    /// `group` may be an integer corresponding to the index of the
+    /// capture group (counted by order of opening parenthesis where `0` is the
+    /// entire match) or it can be a name (consisting of letters, digits or
+    /// underscores) corresponding to a named capture group.
+    ///
+    /// If `group` isn't a valid capture group (whether the name doesn't exist
+    /// or isn't a valid index), then it is replaced with the empty string.
+    ///
+    /// The longest possible name is used. e.g., `$1a` looks up the capture
+    /// group named `1a` and not the capture group at index `1`. To exert more
+    /// precise control over the name, use braces, e.g., `${1}a`.
+    ///
+    /// To write a literal `$`, use `$$`.    
+    pub fn expand(&self, replacement: &str, dst: &mut String) {
+        let mut cursor = Cursor::new(mem::take(dst).into_bytes());
```

Comment:
To make this work on Rust 1.32:

```suggestion
        let mut cursor = Cursor::new(mem::replace(dst, String::new()).into_bytes());
```

Reply:
For that method names, I don't like mashing two verbs together. How about `expansion`, `append_expansion`, and `write_expansion`? It's kind of unfortunate that the actual "append" method for string is called `extend`; `extend_expansion` doesn't make sense and something like `extend_with_expansion` just seems way too long.

Reply:
I uploaded a new patch with the names I proposed, and added `Expander::quote`, since that's another feature I'm familiar with from Python.

The `set_strict` method could be eliminated if you don't think reporting errors in the expansion template is worthwhile. In that case, `expansion` could return `String` and `append_expansion` could return `()`.